syntax = "proto3";

message empty_request {
    string protocol_version = 1;
}

message getif_response {
    message if_info_entry {
        string if_name = 1;
        string if_description = 2;
    }
    string protocol_version = 1;
    repeated if_info_entry interface_list = 2;
}

service tele_service {
    rpc get_if_list(empty_request) returns(getif_response);
}